src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs

```diff
@@ -8,19 +8,29 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class CoreClrAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
     {
-        private readonly AssemblyLoadContext _loadContext;
+        private AssemblyLoadContext _loadContext;
 
         public CoreClrAnalyzerAssemblyLoader()
         {
-            _loadContext = AssemblyLoadContext.GetLoadContext(typeof(CoreClrAnalyzerAssemblyLoader).GetTypeInfo().Assembly);
+        }
 
-            _loadContext.Resolving += (context, name) =>
+        protected override Assembly LoadFromPathImpl(string fullPath)
+        {
+            // Initialize the _loadContext in lazy fashion
+            if (_loadContext == null)
```

Comment:
It seems odd that we would call the constructor on platforms that don't have ALC. I'd prefer revisiting the caller of the .ctor instead

---
In reply to: [125975335](https://github.com/dotnet/roslyn/pull/20656#discussion_r125975335) [](ancestors = 125975335)